### PR TITLE
RFC: demote "cockroach sql" (but perhaps not remove it entirely)

### PR DESCRIPTION
**Pull request by knz**
[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/4059)

### CODE REVIEW

**Comment by knz** [comment]
@bdarnell I took your suggestion to look at pgcli on board, however pgcli queries a lot of pg-specific stuff from a db upon connecting to it and thus will not work out of the box. In contrast psql does, already.

**Comment by petermattis** [comment]
Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions.

---

*[docs/RFCS/remove_sql_cli.rst, line 73 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4059#-K9SkchiNEt2N-UUa6T9:-K9SkchiNEt2N-UUa6TA:-2139590679) ([raw file](https://github.com/cockroachdb/cockroach/blob/7af8738fcdac1ff029ab8ec8e8bc8e804ade83d3/docs/RFCS/remove_sql_cli.rst#L73)):*
I'm fine with this, though it is low priority documentation work (in my opinion).

---

*[docs/RFCS/remove_sql_cli.rst, line 77 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4059#-K9SlFFESFGHjcqMC63R:-K9SlFFESFGHjcqMC63S:-1002253436) ([raw file](https://github.com/cockroachdb/cockroach/blob/7af8738fcdac1ff029ab8ec8e8bc8e804ade83d3/docs/RFCS/remove_sql_cli.rst#L77)):*
I feel strongly that we shouldn't do this. We're implementing a SQL database, we should provide a SQL shell with our product. It is pretty awesome that we're compatible with `psql` and any other SQL shell that can utilize the pgwire protocol, but that doesn't mean we shouldn't provide our own. Yes, our SQL shell has bugs and limitations right now, but those can be fixed.

---

*[docs/RFCS/remove_sql_cli.rst, line 79 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4059#-K9SkPdn_tM3Nd4hmoya:-K9SkPdn_tM3Nd4hmoyb:1447979184) ([raw file](https://github.com/cockroachdb/cockroach/blob/7af8738fcdac1ff029ab8ec8e8bc8e804ade83d3/docs/RFCS/remove_sql_cli.rst#L79)):*
We should probably do this regardless of the decision on this RFC as part of our pgwire compatibility testing. 

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4059)*

**Comment by tschottdorf** [comment]
nothing to add to @petermattis' comments.

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4059#-:-K9SyZYGk8g-Hx1URWf1:1289513278)*

**Comment by bdarnell** [comment]
I agree with Peter; we should ship our own client (for branding purposes and minimizing user confusion if nothing else). I think the main issues with our current client should be easy and worthwhile to fix. We may not have as many features as other clients at first but that's OK. A custom client also gives us a place to put commands that are not (currently) exposed via SQL, like splits, and gives us the opportunity to switch from pgwire to another protocol when/if that becomes desirable. 

I'd be OK with forking and extending some existing client if there is a suitable one. We don't have to build everything from scratch as long as the finished product is under our control (I  wouldn't want to count on the PostgreSQL team making the behavior of `\d` configurable in `psql`, for example). I was hoping `pgcli` would be a good basis since they have both mysql and postgres clients, but it doesn't look like there's actually much shared code between the two. 

---

Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[docs/RFCS/remove_sql_cli.rst, line 48 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4059#-K9TFzJP3GrN68fPk4Ij:-K9TFzJP3GrN68fPk4Ik:714611770) ([raw file](https://github.com/cockroachdb/cockroach/blob/7af8738fcdac1ff029ab8ec8e8bc8e804ade83d3/docs/RFCS/remove_sql_cli.rst#L48)):*
I believe the `\copy` feature uses parts of the protocol that we don't implement. 

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4059#-:-K9TEHBbuqVK6wsmIFta:1490472847)*

File: docs/RFCS/remove_sql_cli.rst [hunk 1]
```diff
@@ -0,0 +1,100 @@
+:Feature Name: Supersede the SQL command-line client with psql
```
Comment (bdarnell) [inline docs/RFCS/remove_sql_cli.rst]: Please convert this to markdown; as I said in #4056 I don't think we need two different formats for RFCs.

**Comment by knz** [comment]
Review status: 0 of 1 files reviewed at latest revision, 5 unresolved discussions.

---

*[docs/RFCS/remove_sql_cli.md, line 1 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4059#-K9SgsSW5oGnpdcljjMz-r1-1:-K9XvpCIy1VHOW0-IWcE:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/7af8738fcdac1ff029ab8ec8e8bc8e804ade83d3/docs/RFCS/remove_sql_cli.md#L1)):*
Done.

---

*[docs/RFCS/remove_sql_cli.md, line 79 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4059#-K9SkPdn_tM3Nd4hmoya:-K9Xw8hunxOSGnYb_BKF:1909536523) ([raw file](https://github.com/cockroachdb/cockroach/blob/7af8738fcdac1ff029ab8ec8e8bc8e804ade83d3/docs/RFCS/remove_sql_cli.md#L79)):*
Ok filed #4083.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4059)*

**Comment by knz** [comment]
So it looks like this one should get status "rejected", right?
Do we still merge it with that new status or forget it entirely?

**Comment by petermattis** [comment]
Let's mark it as rejected and merge it.

**Pull request merged**